docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,597 @@
+* For metrics, one of `metric_names`, [resource][] must be specified with a valid non-empty value for a valid configuration. The `span_names`, `span_kinds`, `log_bodies`, `log_severity_texts`, `log_severity_number` and `services` attributes are invalid.
+
+For `metric_names`, a match occurs if the metric name matches at least one item in the list.
+For `span_kinds`, a match occurs if the span's span kind matches at least one item in this list.
+
+### regexp block
+
+This block is optional configuration for the `match_type` of `"regexp"`.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`cacheenabled` | `bool` | Determines whether match results are LRU cached. | `false` | no
+`cachemaxnumentries` | `int` | The max number of entries of the LRU cache that stores match results. | `0` | no
+
+Enabling `cacheenabled` could make subsequent matches faster.
+Cache size is unlimited unless `cachemaxnumentries` is also specified.
+
+`cachemaxnumentries` is ignored if `cacheenabled` is false.
+
+### attribute block
+
+This block specifies a list of attributes to match against.
+Only `match_type = "strict"` is allowed if `attribute` is specified.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`key` | `string` | The attribute key. | | yes
+`value` | `any` | The value to match against. | | no
+
+If `value` is not set, any value will match.
+The type of `value` could be a number, a string or a boolean.
+
+### resource block
+
+This block specifies items to match the resources against.
+A match occurs if the input data resources matches at least one `resource` block.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`key` | `string` | The attribute key. | | yes
+`value` | `any` | The value to match against. | | no
+
+If `value` is not set, any value will match.
+The type of `value` could be a number, a string or a boolean.
+
+### library block
+
+This block specifies items to match the implementation library against.
+A match occurs if the span's implementation library matches at least one `library` block.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`name` | `string` | The attribute key. | | yes
+`version` | | The value to match against. | | yes
+
+### log_severity_number block
+
+This block defines how to match based on a log record's SeverityNumber field.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`min` | `int` | The lowest severity that may be matched. | | yes
```

Comment:
Not sure what is the best way to document the allowed [plog values](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.64.0/pdata/plog/logs.go#L76). I think I'll just copy all the values here, so that we don't link to the otel code.

Reply:
Yes, you should copy the values. But also, I think we should define our own string-based mapping here using encoding.TextMarshaler and encoding.TextUnmarshaler because `min = 7` is going to confuse people.